libsignal-protocol/src/crypto/openssl.rs

```diff
@@ -38,20 +39,36 @@ impl OpenSSLCrypto {
             (SignalCipherType::AesCbcPkcs5, 32) => Cipher::aes_256_cbc(),
             _ => unreachable!(),
         };
-
         let block_size = signal_cipher_type.block_size();
-        let mut result = Vec::with_capacity(data.len() + block_size);
```

Comment:
IMPORTANT :warning:

Reply:
Hmm, that trailing padding of 4 bytes is weird. Could it have something to do with being a block cipher and only emitting data in multiples of X bytes?

Reply:
turns out that, the crypter's `update` and `finalize` methods, return the `count` of bytes written in the buffer so far, so at the end of the `encryption/decryption` you should `truncate` the result buffer to that `count` of buffer.
by doing so, now it has the same result as `native` impl.